crates/transport-ipc/src/lib.rs

```diff
@@ -193,9 +193,15 @@ where
 
             // read more data into the buffer
             match ready!(poll_read_buf(this.reader.as_mut(), cx, &mut this.buf)) {
+                Ok(0) => {
+                    // stream is no longer readable and we're also unable to decode any more
+                    // data. This happens if the IPC socket is closed by the other end.
+                    // so we can return `None` here.
+                    debug!("IPC socket EOF, stream is closed");
```

Comment:
would consider making these traces

Reply:
can do eventually, but for now this is useful on debug